### PR TITLE
Easier dev-setup

### PR DESCRIPTION
**Pull request by mixman**
Allows running project with ```python3 run.py```.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,21 +11,6 @@ The DOC file has some explanations about the structure & features.
 ```
 
 
-# Dev Setup:
```
Comment (MihaiFtr) [inline README.md]: I'd leave these instructions at the top as the secrets/local_settings must be put in by hand before running vagrant up.

File: README.md [hunk 2]
```diff
@@ -37,26 +22,29 @@ https://dev.futurice.com:8081/vimma/
 
 ## Running on your machine:
 
-Create a Python3 virtual enviroment and use it for most commands.
```
Comment (MihaiFtr) [inline README.md]: I wouldn't lose the commands below showing exactly what and how to run. The other stuff (vagrant, run.py) are conveniences for these commands.
Should we move them to their own section “doing everything by hand” at the end?

File: run.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import os, subprocess, locale
+cwd = os.getcwd()
+def cmd(execute):
+    return subprocess.call(execute.format(cwd).split())
```
Comment (MihaiFtr) [inline run.py]: Splitting on whitespace is error-prone. Let's use `subprocess.call([arg1, arg2, argN])` instead.

File: scripts/dev-db-reset.py [hunk 4]
```diff
@@ -31,7 +36,14 @@ def populate():
 
 
 if __name__ == '__main__':
-    parse_args()
+    args = parse_args()
+    if args.keep:
+        from vimma.models import Permission
+        try:
+            Permission.objects.all()
```
Comment (MihaiFtr) [inline scripts/dev-db-reset.py]: Let's revert the changes to this file and instead check if the DB exists in run.py – that seems the proper place for this.
This is turning from a straightforwrd “throw away and re-create the database” into “delete and recreate, or do nothing with -keep, except if the permissions are missing”.

As an aside, this file is just a helper for setting up dev, unused in production.

File: vimmasite/vimma/celeryconfig.py [hunk 5]
```diff
@@ -5,6 +5,10 @@
 _every_5min = crontab(minute='*/5')
 _every_1h = crontab(minute=0)
 
+BROKER_BACKEND = "redis"
```
Comment (MihaiFtr) [inline vimmasite/vimma/celeryconfig.py]: Why use redis, instead of the default rabbitmq? Was there some problem getting it running?

File: README.md (hunk 2 above)
Comment (mixman) [inline README.md]: We can move to the bottom. The point is to move away from "doing everything by hand".

File: run.py (hunk 3 above)
Comment (mixman) [inline run.py]: Sure. Can refactor this to be more bullet-proof and nice with a proper library later on (that wouldn't be call[...])

File: scripts/dev-db-reset.py (hunk 4 above)
Comment (mixman) [inline scripts/dev-db-reset.py]: I've already done this, and changed the filename, to be importable.

I'm using your suggestion for checking if the database is there or not with the Permission -model query.

File: scripts/startup.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+if not os.path.isfile('./config/secrets.py'):
+    cmd('cp {}/config/secrets.py.example ./config/secrets.py')
+if not os.path.isfile('./config/dev_secrets.py'):
+    cmd('cp {}/config/dev_secrets.py.example ./config/dev_secrets.py')
+
+
+if not os.path.isdir('./env'):
+    cmd('virtualenv -p python3 env')
+
+def shell_source(script):
```
Comment (MihaiFtr) [inline scripts/startup.py]: Same comment on whitespace splitting for subprocess.call.

Also instead of sourcing `env/bin/activate` (and the `shell=True` and environment updates) we can simply prepend `env/bin/` to the PATH environment variable. That's supposed to be the only goal of the `activate` script, and I've used the PATH approach successfully on Linux.

File: README.md (hunk 2 above)
Comment (MihaiFtr) [inline README.md]: Agreed – I haven't been running these by hand for a long time. Just trying to future-proof for the day when our convenience helpers won't work and no one remembers these commands :)

File: run.py (hunk 3 above)
Comment (MihaiFtr) [inline run.py]: Just for straightforward things like this I wouldn't add additional libraries (which may also hide other bugs).

File: vimmasite/vimma/celeryconfig.py (hunk 5 above)
Comment (mixman) [inline vimmasite/vimma/celeryconfig.py]: One less daemon to run and install. Redis is more often used than Rabbitmq (atleast in the projects I've done). Rabbitmq can be used for production still.

Comment (MihaiFtr) [inline vimmasite/vimma/celeryconfig.py]: We'd be replacing one server with another, and tweaking the working project setup. I'd keep rabbitmq, it's worked well so far (dev & prod).

Comment (mixman) [inline vimmasite/vimma/celeryconfig.py]: You can. :) I'll check how local_settings and celeryconfig play together.

File: README.md [hunk 7]
```diff
@@ -35,31 +35,61 @@ vagrant up
...
+ln -s /usr/bin/nodejs /usr/bin/node
+
 virtualenv -p python3 env
 . env/bin/activate
 pip install -r req.txt
 
 npm install
-PATH=`pwd`/node_modules/.bin:$PATH ./scripts/bower-reset.py	# bower install
+cd vimmasite/vimma/static/vimma/components
+bower install
```
Comment (MihaiFtr) [inline README.md]: This and the apt-get commands require everything to be installed globally. I'd leave the manual instructions minimal and local (not global/root). Procboy could have its separate section, like Vagrant has, as they are both alternatives to the manual way.
